### ISSUE TITLE
Discourse: Evaluation Checklist

### ISSUE THREAD

**Issue by RichardLitt** [ipfs/community#213]
We've had a [trial Discourse forum](http://ipfs.trydiscourse.com) up for over a month. There have been some discussions on it by a few contributors, and at this point, everyone should have had the chance to interface with the forum and see how it works for them. 

Moving forward, I would like to come to a consensus about whether we should either go with Discourse as a possible alternative to our GitHub repo system, or not. **In this thread**, I'd like to establish any concerns people have that we can actionably deal with, and to have a checklist for things which need to be dealt with before we move forward. 

These are some concerns I have heard.

### Have more contributors participate on the forum.

We haven't had enough people buy in.

To test this, we should:
- [ ] Actively move some more discussions over there. **Does this sound right? Which issues? Other ideas?**

### Have a technical discussion on the forum.

Right now it is only meta topics, and we can't evaluate how it would look to have a technical discussion there given that.

To test this, we should:
- [ ] Take a technical issue, and talk about it there. **Do you have suggestions for a good issue?**

### Answer support questions there. 

To test this, we should:
- [ ] Direct users away from ipfs/support for now, and send them to Discourse.

### See whether the scoping of topics makes sense.

Are the categories accurate, and do they help us navigate the board? Are they better than ipfs/support, ipfs/faq, and ipfs/community for saying where things should be posted?

To test this, we should:
- [ ] Have a discussion about it, here. 

### Make sure that everyone has time to see it.

Not everyone has had the time to interface with it. We should prioritize this over code before the end of the year.

To test this, we should:
- [ ] **Get a list of people who feel they haven't had time. Is this you?**
- [ ] Have them scope out some time to look at it, at the cost of other priorities. 

## Migration Plan

These are tasks which need to be done before we can move fully to Discourse, if that is what we want to do.

- [ ] Copy all ipfs/faq issues over here.
- [ ] Build a document that coalesces the 'answered' FAQ questions into a single document. Post it on the website.
- [ ] Mention Discourse on ipfs/ipfs, ipfs/community, ipfs/pm.
- [ ] Close some of the open long-standing issues on ipfs/community which are served by this repo.
- [ ] Close ipfs/support, close all issues, and say that it is now Deprecated. Actively gut the README. Reopen open issues on Discourse.
- [ ] Post a link to Discourse on the IRC channel's topic.
- [ ] Announce on Twitter, Reddit, and the blog.

## Anything else?

If you have other concerns, can you list them below? Comments, too. Hopefully, we can grow this list out, make it better, and actively check off items and come to conclusions about each concern. 

_If you feel this could be phrased better, please let me know. Always happy to improve._

**Comment by RichardLitt** [comment]
On moving FAQ and Support to Discourse

While it is possible to disable issues for a repository temporarily, and bring those issues back later as needed, this breaks links to those issues elsewhere. One thing we could do immediately is to add an issue template to those Repositories, like we've done with IPFS/IPFS now (thanks @whyrusleeping), asking users to post in Discourse. @whyrusleeping pointed out to me that we could also do this without being harsh; just ask that they start posting here. A possible downside to this plan is that now we would need to check both Discourse and ipfs/support and ipfs/faq, simultaneously.

**Comment by flyingzumwalt** [comment]
Could we set up the subdomain sooner rather than later? I'd prefer to use a stable url when recruiting people to participate in the tests. That way the link won't break if/when we move to adopt Discourse long-term.

**Comment by kevina** [comment]
As the site stands now I won't use it.  I find text hard to read due to lack of contrast.  Please see http://contrastrebellion.com/ and https://www.nngroup.com/articles/low-contrast/.  If the background is white I prefix pure black for the main text, but I am okay with a _slightly_ lighter color (GitHub `#333` is probably about as light as I will tolerate).  If I am forced to use the site as it stands I will have to install a browser extension to tweak the style sheet so I can read it with straining my eyes.

I am not 100% sure but there also doesn't seam to be a way to get email on your own replies.  I find this feature extremely useful, but since many other forms don't provide it something I can live with.

**Comment by flyingzumwalt** [comment]
We have complete control of the styling, so we can fix the contrast. I agree that high contrast is a basic accessibility issue that we have to prioritize. The current style is a quick and dirty imitation of the CSS from the IPFS.io website.

cc @dignifiedquire

**Comment by dignifiedquire** [comment]
I've just updated the base text color to be `#333`. It might take a couple of refreshes before your cache updates though.

**Comment by kevina** [comment]
Thanks!  That's better.  Note that many other web sites now seam to use `#222`, that might be a bit better.  But as it is now is okay for me.

**Comment by lidel** [comment]
@kevina: 
> I am not 100% sure but there also doesn't seam to be a way to get email on your own replies. I find this feature extremely useful, but since many other forms don't provide it something I can live with.

You can enable "[Mailing list mode](https://meta.discourse.org/t/what-is-mailing-list-mode/46008/2?u=lidel)" in _Preferences_  
("gear" icon after you click on your avatar):

> ![2016-12-18-003019_3440x1440_scrot](https://cloud.githubusercontent.com/assets/157609/21290405/6bdf068e-c4b9-11e6-95bb-4631d52c307e.png)

>  ![2016-12-18-003550_495x82_scrot](https://cloud.githubusercontent.com/assets/157609/21290423/f4a0566c-c4b9-11e6-95f3-e49e3ca85488.png)

**Comment by kevina** [comment]
@lidel I saw that, it is unclear if it will enable emails to be sent on your own activity like GitHub now allows.  From the limited testing I saw in try.discourse.org it does not, at least for private messages.  I can't post to the Sandbox because my account is "on hold".  So I waited, and then when I came back my account seams to have been deleted.

**Comment by flyingzumwalt** [comment]
Marta posted a bunch of great questions on Discourse yesterday. We should use these to test Discourse. @RichardLitt can you drum up people to respond -- both from #ipfs and #ipfs-dev -- and watch the discussions?

**Comment by VictorBjelkholm** [comment]
Have we made any progress on this? Seemingly, the activity rate on Discourse is not that high (last post seems to have been at 23th of January)

**Comment by lidel** [comment]
Low participation is not surprising:
a)  `ipfs/faq`, `ipfs/support` and `ipfs/community` are still on Github, people are not directed to Discourse because there is not good content there (chicken and egg problem)
b) invasive anti-spam settings I mentioned [here](http://ipfs.trydiscourse.com/t/discourse-annoyances/35/18) may discourage people from posting at Discourse (there are no restrictions at Github, you just create account and post anything you want)

**Comment by RichardLitt** [comment]
I agree with point a), at least.

I am unable to continue pushing this forward. I'll leave this up to the rest of the IPFS team to enact.

**Comment by lgierth** [comment]
We could set up a two-way sync between discourse and github. Or just move everything over and close the repos (I'd be down for that by now).

**Comment by dignifiedquire** [comment]
I think we should

1. move everything that makes sense over
2. have a DRI for discourse handling

**Comment by lidel** [comment]
Having _https://github.com/ipfs/community/pull/246: Decision Document: Creating a Community Forum for IPFS and Related Projects_
what will be the next step? 

Changing `http://ipfs.trydiscourse.com` to  `*.ipfs.io` domain picked in https://github.com/ipfs/ops-requests/issues/33?

**Comment by flyingzumwalt** [comment]
I just checked in with @lgierth about this. ipfs.io domain only allows https, even on subdomains. That means we will only be able to move discourse to discuss.ipfs.io after spinning up our own instance. We were planning to do that anyway. It just means we will have to wait a few more days before making the switch.

What are the other next steps? Off the top of my head:
- rename from "discourse trial" to IPFS forum
- after setting up the new subdomain, announce on all channels
- do the steps outlined in the proposal on loomio

**Comment by diasdavid** [comment]
# It has been done https://discuss.ipfs.io/ 🎉
